source/extensions/quic_listeners/quiche/platform/BUILD

```diff
@@ -167,10 +167,31 @@ envoy_cc_library(
         "//source/common/filesystem:directory_lib",
         "//source/common/filesystem:filesystem_lib",
         "//source/common/http:utility_lib",
-        "@com_googlesource_quiche//:quic_core_types_lib",
     ],
 )
 
+envoy_cc_library(
+    name = "quiche_common_platform_impl_lib",
+    hdrs = [
+        "quiche_logging_impl.h",
+        "quiche_ptr_util_impl.h",
+        "quiche_unordered_containers_impl.h",
+    ],
+    external_deps = [
+        "abseil_hash",
+        "abseil_node_hash_map",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [":quic_platform_base_impl_lib"],
```

Comment:
add `:quic_platform_logging_impl_lib` to fix clang-tidy, bazel doesn't build header files only target explicitly but clang-tidy checks whether you added dependencies correctly.

Reply:
Woops, just realized another problem that quic_logging_impl.h is duplicated in both quic_platform_logging_impl_lib and quic_platform_base_impl_lib. 

Fixed. Thanks for looking into this problem!

Reply:
Still not fixed. This is really weird. Looking at the errors a few lines above:
Error while processing /source/source/extensions/quic_listeners/quiche/platform/quic_containers_impl.h.
10294 warnings and 1 error generated.
Error while processing /source/test/extensions/quic_listeners/quiche/platform/quiche_test_impl.h.
16998 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/quiche_logging_impl.h.
23702 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/quiche_logging_impl.cc.
31722 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/spdy_containers_impl.h.
37223 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/spdy_string_utils_impl.h.
39075 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/quiche_ptr_util_impl.h.
43844 warnings and 1 error generated.
Error while processing /source/source/extensions/quic_listeners/quiche/platform/quiche_unordered_containers_impl.h.
warning: do not reference any 'internal' namespaces; those implementation details are reserved to Abseil [abseil-no-internal-dependencies]
bazel-out/k8-fastbuild/genfiles/external/com_googlesource_quiche/quiche/common/platform/api/quiche_logging.h:4:10: error: 'extensions/quic_listeners/quiche/platform/quiche_logging_impl.h' file not found [clang-diagnostic-error]
#include "extensions/quic_listeners/quiche/platform/quiche_logging_impl.h"

The file path is `/source/source/extensions/quic_listeners/quiche/platform/...` or `/source/test/extensions/quic_listeners/quiche/platform/...`. There is a `/source` added to the actual file path. 

Why would clang_tidy CI do that?

Reply:
CI image checkout repository to `/source` so the path is correct.